launcher/src/main/kotlin/com/cognifide/gradle/aem/launcher/Launcher.kt

```diff
@@ -142,7 +144,9 @@ class Launcher(val args: Array<String>) {
 
         const val ARG_SAVE_PREFIX = "-P"
 
-        val ARGS = listOf(ARG_SAVE_PROPS, ARG_PRINT_STACKTRACE, ARG_NO_COLOR_OUTPUT, ARG_WORK_DIR)
+        val ARGS = setOf(ARG_SAVE_PROPS, ARG_PRINT_STACKTRACE, ARG_NO_COLOR_OUTPUT, ARG_WORK_DIR)
+
+        val WRAPPER_ARGS = setOf("--gradle-version", "--distribution-type", "--gradle-distribution-url", "--gradle-distribution-sha256-sum")
```

Comment:
this is what I dislike at most. hardcoding that stuff here. but if there is no option to avoid that... fine.

Reply:
I think I can’t be avoided. The arguments are in annotations in the `Wrapper` class:

https://github.com/gradle/gradle/blob/94d41f583f67a9e51a726501dc3332cd63e6d0af/subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java#L305

and that’s not part of gradle-api.jar that we have as a dependency.